[12.0][MIG] Bank transfer

simahawk opened the pull request:
Opening this here to not mess up while https://github.com/shopinvader/odoo-shopinvader-payment/pull/7 is still WIP.

I've tested it locally but seems there's something broken ATM. 
I have to debug it.

lmignon committed:
[IMP] Add bank transfer payment

lmignon committed:
[IMP] Add bank transfer payment

simahawk committed:
Bank transfer: migrate to 12.0

sbidoul commented on invader_payment_bank_transfer/services/payment_bank_transfer.py:
```diff
@@ -0,0 +1,46 @@
+        )
+        transaction = self.env["payment.transaction"].browse()
+        payable._invader_payment_start(transaction, payment_mode)
+        payable._invader_payment_success(transaction)
```
We must work on this. I know `_invader_payment_success` is here to trigger the cart confirmation. Yet it's a bit strange to say the payment is successful. @lmignon

sbidoul replied:
Also don't we want to create a `payment.transaction`?

simahawk replied:
@sbidoul I'm just getting familiar w/ this but to me seems reasonable that we don't create a transaction here: there's no transaction yet. Transaction will be added manually via backend as soon as the payment is received.
Am I missing anything?

We could argue that w/ manual payment the cart / order should be in another state, eg: "Waiting payment".

sbidoul replied:
About `payment.transaction` there are two reasons to create it IMO:
- standard Odoo does it (it creates a transaction in `pending` state), and we want to be as close as possible to Odoo standard here
- it could become something where we hook some automation when the bank transfer is confirmed; if we don't create the transaction we don't have much trace that a payment operation was intended

simahawk replied:
I see. Then maybe you are right. I'll check.

sbidoul replied:
> We could argue that w/ manual payment the cart / order should be in another state, eg: "Waiting payment"

I would not add a new SO state for that. The payment status can be deduced from the transaction statuses.

sbidoul replied:
So following #4 you can now call `_invader_payment_accepted` instead of `_invader_payment_success`. See also some additional thoughts in the docstrings in #4.

sbidoul commented:
@simahawk my remarks do not prevent merging this. Is it good enough for merging?

simahawk commented:
hmm I don't know. Locally it fails somehow. I still have to test it properly (but as you know I'm having some issues w/ Locomotive to run locally :wink: )

simahawk committed:
Bank transfer: add readme

simahawk committed:
Fix invader_payment: restore base component

Without this all payment handlers are broken
as they rely on `invader.payment` generic service.

simahawk committed:
Linting

simahawk commented:
rebased on #4 , fixed component, adapted to api change: tested and worked.

sbidoul commented:
Thanks!
